### PR TITLE
Adds a status effect to try to mitigate sniper stacking

### PR DESCRIPTION
**Pull request by ivanmixo**
## About The Pull Request
Basically the way this works is:

- Xeno gets hit with a projectile from a sniper rifle
- Xeno gets a hidden status effect applied that will make all subsequent impacting sniper rounds do 10% of their original damage for the duration of the firing gun's `fire_delay` duration (-1 to make shots on "cooldown" not take the 90% damage malus)

To illustrate, lets say a marksman round from a DMR-37 hits a crusher. The crusher will gain the status effect on impact, and while she has the status effect lets say another sniper round hits her. This subsequent round will only do 10% of the damage it would usually do. After the original DMR-37's fire delay has passed the status effect will have worn off and the crusher will once again be open to take full damage from any further sniper rounds. 

Note however that all sunder, CC, shatter and what have you from the subsequent rounds will still apply, so it isn't totally worthless to still fire your shot, although you'll want to avoid shooting at targets other marines are firing at to maximize damage.

Current:

https://github.com/user-attachments/assets/a05a16d7-2850-470c-9d49-08c336a38fdc

With proposed changes:

https://github.com/user-attachments/assets/e0a6c986-b384-4354-9d00-1b7258fbe306

Single DMR firing for reference

https://github.com/user-attachments/assets/057b8c6d-42da-4b9a-b2e0-e7458d0b9581

I've also renamed the flag previously named `AMMO_SNIPER` to something more appropriate. 
## Why It's Good For The Game

Sniper stacking has long been an issue, killing xenos from offscreen within possibly less than a second. While this may require *some* coordination I'd wager it's a bit too much reward to be able to instakill anything that comes on your screen just because you got some buddies to stack with you. This same logic doesn't apply to other guns because the xeno can clearly see the amount of marines they'll be engaging with and therefore able to avoid the fight altogether until they can find a better moment to strike. 

This should hopefully help alleviate some of the issues with offscreen gameplay that has been occurring lately by limiting its effectiveness while also not gimping solo snipers since they don't pose that large an issue. If this proves to be too severe or too lax for whatever reason, we can just adjust the numbers. 
## Changelog
:cl:
balance: Made some changes to curb stacking multiple snipers, check the PR for a more verbose explanation
/:cl:

### CODE REVIEW

**Comment by FosterGrime-Heymynameslouie** [comment]
10% of og damage only being applies seems a little bit too harsh? most I see sniper stacking is just 1 to 2 dudes shooting the same target and given how most xeno interaction have them pop in and out and how walls/resinwalls/map setups usually happens youll only ever have a single target at a time and only dealing 10% and what usually is your only target at the time seems a bit too harsh. Maybe like 20 or 25 would be alot more cooler?

I do know the intent of the pr is to mitigate like, 5 to 6 guys targetting the same xeno and it works well on that regard but it seems to  disproportionately hurt small scale stacking of 1 to 2 (maybe 3?). 

Probably a cop out but I think having shots progressively have lower effectiveness overtime seems like a better alternative than a flat percentage. Like the first shot and the shoots from the same shooter will always be a 100% effective but subsequent shots  from sniper rifles from different people get progressively less and less effective? Seems counterintuitive but I think its worth a shot given how offscreens are generally unfun to fight against. 

cool pr

**Review by Kuro020 — APPROVED** [review]
like the idea, numbers might need tweaks, but well see!

**Comment by Pariah919** [comment]
10% is pretty brutal. Should do like 20%.

**Comment by SimplySmiley123** [comment]
Not sure hidden status effects are good, imagine someone who isn't constantly diving into github trying to figure out why his gun is suddenly gimped out of the blue for no visible reason.

**Comment by Technical-l** [comment]
10% seems really rough and i'd imagine the hidden status effect to be really annoying (especially for new players or those who don't follow updates). Donno the best way to do it but having some in-game way of letting people know of this mechanic would be cool

**Comment by DrAmazing343** [comment]
ided

as someone else said, we don't have the 2x2 shoulder fire sniper cubes rolling out to the field and the "coordinated" effect of sniper stacks is mostly due to the chokepoint nature of xeno gameplay. this is going to make sniping extraordinarily unrewarding unless you are literally the only sniper firing at a target... because xenos need to be handheld even harder?

like, I know I'm shitposting a little, but seeing dchat mald over nothing really does christen this as an ided that I think has no place in the current tgmc.

**Comment by iforgotmeword** [comment]
just something i thought of something
this shouldn't effect if the xeno is on screen
the status effect should only be when they are shot off screen

**Comment by DorfGuy1** [comment]
This is a horrible idea. Consider the overall balance.  You might be able to snipe down a boiler or a floaty boi in a few seconds, if everyone is sweating and coordinating in discord or some shit, but I doubt it. Plus, do xenos really need yet more buffs?

**Comment by Sun-Soaked** [comment]
> This is a horrible idea. Consider the overall balance. You might be able to snipe down a boiler or a floaty boi in a few seconds, if everyone is sweating and coordinating in discord or some shit, but I doubt it. Plus, do xenos really need yet more buffs?

https://github.com/user-attachments/assets/d0ed3bfc-98a7-4335-831d-52953c876bdb

**Comment by chemistrymain3** [comment]
reduce the debuff to like 50% or something

**Comment by ButterChugger** [comment]
>Remove aim mode from main guns
>Keep aim on things in marskman and sidearms "to offer a niche"
>Oh wait, remove it from the battle rifle. No need for automatic fire for aiming that isn't a pistol. 
>Everyone begins to use the semi automatic sniper rifle to aim due to removal. 
>Some benos get hit by what is a colonial firing line because ungas cannot spray anymore. 
>REE NERF AIM MODE MORE

This will be pushed regardless. Still disgusting.

**Comment by CaputCapital** [comment]
> > Remove aim mode from main guns
> > Keep aim on things in marskman and sidearms "to offer a niche"
> > Oh wait, remove it from the battle rifle. No need for automatic fire for aiming that isn't a pistol.
> > Everyone begins to use the semi automatic sniper rifle to aim due to removal.
> > Some benos get hit by what is a colonial firing line because ungas cannot spray anymore.
> > REE NERF AIM MODE MORE
> 
> This will be pushed regardless. Still disgusting.

A PR for mitigating xeno stuns, making it so that chain-stuns are less effective would never see the light of day, so I don't know why this is in development.

As TGMC you're supposed to work together and coordinate but lately and long before PRs like this, it's clear that ivan & lumi want to discourage certain methods of teamwork and alternative playstyles. From killing shield straps over petty reasons (no we want you to use deployable shields the way WE see fit!), to killing pistol seasonals because marines did well with them one round, to killing laser snipers all because a few robots shitting on xenos using ricochet mode. I don't know what else to say other than to start using other weapons and martini henry's, that is if they don't nerf it into oblivion even after Kuro gatekeeps aimmode on it.

TL;DR: Fuck sniper rifles you're better off just using virtually anything else at this point since they'll be merging this soon.

**Comment by novaepee** [comment]
https://github.com/user-attachments/assets/0b6ee922-ffac-4e8e-b917-0f8c1645e950

**Comment by CaputCapital** [comment]
> TGMC_WOWZA_KILLS.mp4

Queen overextended. XOB and acid maw would've made the Minhs scatter like ants.

**Comment by DrAmazing343** [comment]
> > This is a horrible idea. Consider the overall balance. You might be able to snipe down a boiler or a floaty boi in a few seconds, if everyone is sweating and coordinating in discord or some shit, but I doubt it. Plus, do xenos really need yet more buffs?
> 
> ripcrusher.mp4

the way I straight up be Eating My Words Right Now???

I think this is still a sort of unintuitive solution but a necessary one at this point

**Comment by ButterChugger** [comment]
Actually this brings back into higher perspective of what aim mode is. Fairly certain speaking many people want it, and given the fact that the only option of it ARE these high powered, armor piercing, sunder inducing rifles benos are now feeling the burn of what it's like when nothing but HIGHER POWER guns firing at once. By forcing users to who don't want to conform to stand next to each other to shoulder fire you've introduced a problem that was at best, spare before removal and now shoved front and center. Probably best to put aim on guns of LOWER CALIBER, or at bare minimum, flank.  

>TGMC_WOWZA_KILLS.mp4
A literal line of bolt action, non semi automatic sniper rifles firing. Clearly joking, clearly not normal play.

**Comment by Aporhtonoma** [comment]
instead of forcing an arbitrary debuff punishing people with very minimal feedback to players who  are using using the same gun (when its infinetly available and for free) why are we not limiting snipers if they are such a big issue .This change is essentially punishing all sniper players for playing sniper when others play sniper already.

and before the argument amr and autosnipers are req buyable is thrown. So what xeno obs and mortars are buyable and stackable as well they are essentially the same thing but we are punishing only one side because...?

**Comment by sockbyte** [comment]
> > This is a horrible idea. Consider the overall balance. You might be able to snipe down a boiler or a floaty boi in a few seconds, if everyone is sweating and coordinating in discord or some shit, but I doubt it. Plus, do xenos really need yet more buffs?
> 
>  ripcrusher.mp4

In this clip there's an autosniper (req sniper) and two BR-8s (not snipers). This PR wouldn't actually do anything here. 

Although this PR stops the rare sniper stack of 5-6 people where they might actually be able to kill from a distance, it's way too punishing for making only **a single player** be able to use a sniper at a time. If snipers are such a huge issue where having more than 1 is "too OP" they should be limited either in vendor to a role instead of a hidden mechanic where you have no feedback whether your gun is doing damage or not.

**Comment by Woker2004** [comment]
> > This is a horrible idea. Consider the overall balance. You might be able to snipe down a boiler or a floaty boi in a few seconds, if everyone is sweating and coordinating in discord or some shit, but I doubt it. Plus, do xenos really need yet more buffs?
> 
>  ripcrusher.mp4

The huge problem in that clip is the DMR, adding a minimum range to the DMR would truthfully solve that massive issue, and maybe making DMR require a scope to use IFF would be more ideal in my opinion, the other snipers are rarely used, let alone this would most likely make AMR just not be picked anymore

**Comment by ButterChugger** [comment]
>although you'll want to avoid shooting at targets other marines are firing at to maximize damage.
With this merge, it is official that teamwork is now counter productive to being a marine. It was a nice ride.
